### PR TITLE
[nrfconnect] Refactored configuration for the nrfconnect examples

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
nrfconnect platform uses Kconfig overlays and defaults in a way that is not consistent with the nRF Connect SDK recommendations

#### Change overview
* Created Kconfig.defaults that set default configs common for all examples (replaced sample-defaults.conf)
* Created Kconfig.features that allows to configure set of options within single feature using single option
* Replaced approach of adding overlays in example cmake file with the pattern of placing prj.conf files in a configuration/build-target/ location that allows adding config files to the build automatically by the toolchain.
* Created debug and release configuration variants for several examples.

#### Testing
Tested manually builds and basic functionalities, however change should not introduce functional changes, as it's only structural refactor.

### CODE REVIEW

File: config/nrfconnect/chip-module/Kconfig.features [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		Enables QSPI NOR with the set of options
+		configuring pages and buffer sizes.
+
+if CHIP_QSPI_NOR
+
+config NORDIC_QSPI_NOR
+	bool
+	default y
+
+config NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE
```
Comment (Damian-Nordic) [inline config/nrfconnect/chip-module/Kconfig.features:34]: I think it's set by default based on DTS, so can potentially be removed now.

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes

File: config/nrfconnect/chip-module/Kconfig.features (hunk 1 above)
Comment (woody-apple) [reply config/nrfconnect/chip-module/Kconfig.features:34]: @kkasperczyk-no ?

Comment (kkasperczyk-no) [reply config/nrfconnect/chip-module/Kconfig.features]: Yes, that's right, fixed.

**Pull request merged**